src/queries/term-level-queries/regexp-query.js

```diff
@@ -44,6 +44,22 @@ class RegexpQuery extends MultiTermQueryBase {
         return this;
     }
 
+    /**
+     * Allow case insensitive matching or not.
+     * Defaults to false.
+     *
+     * @example
+     * const qry = esb.regexpQuery('name.first', 's.*y')
+     *     .caseInsensitive(true);
+     *
+     * @param {boolean} caseInsensitive
+     * @returns {RegexpQuery} returns `this` so that calls can be chained.
+     */
+    caseInsensitive(caseInsensitive) {
```

Comment:
Could you update the `index.d.ts` file for this?

Also, can we add a note mentioning the version of elasticsearch in which this feature is supported?

Reply:
Done.